### PR TITLE
Duplicated type when fields are named the same as operation types

### PR DESCRIPTION
**Pull request by kamilkisiela**
Related: #1365

### CODE REVIEW

File: packages/utils/plugins-helpers/src/flatten-types.ts [hunk 1]
```diff
@@ -84,13 +86,15 @@ export function flattenSelectionSet(selectionSet: SelectionSetItem[], result: Fl
   selectionSet.forEach((item: SelectionSetItem) => {
     if (isFieldNode(item)) {
       if (item.selectionSet.length > 0) {
+        // here
```
Comment (dotansimha) [inline packages/utils/plugins-helpers/src/flatten-types.ts]: here?

File: packages/utils/plugins-helpers/src/flatten-types.ts [hunk 2]
```diff
@@ -84,13 +86,15 @@ export function flattenSelectionSet(selectionSet: SelectionSetItem[], result: Fl
...
         const model = buildModelFromField(item, result);
         item.type = model.modelType;
         result.push(model);
 
         flattenSelectionSet(item.selectionSet, result);
       }
     } else if (isInlineFragmentNode(item)) {
+      // here
```
Comment (dotansimha) [inline packages/utils/plugins-helpers/src/flatten-types.ts]: here?

Comment (kamilkisiela) [reply packages/utils/plugins-helpers/src/flatten-types.ts]: oopsie

File: packages/utils/plugins-helpers/src/flatten-types.ts (hunk 1 above)
Comment (kamilkisiela) [reply packages/utils/plugins-helpers/src/flatten-types.ts]: oopsie

Comment (kamilkisiela) [reply packages/utils/plugins-helpers/src/flatten-types.ts]:
```suggestion
```

File: packages/utils/plugins-helpers/src/flatten-types.ts (hunk 2 above)
Comment (kamilkisiela) [inline packages/utils/plugins-helpers/src/flatten-types.ts]:
```suggestion
```

**Pull request merged**